otelgorm/option.go

```diff
@@ -49,3 +49,12 @@ func WithoutMetrics() Option {
 		p.excludeMetrics = true
 	}
 }
+
+// WithDryRunSpans will add trace spans for "dry run" callback invocations from
+// Gorm. "Dry Run" invocations occur when Gorm is just rendering SQL, but not
+// actually executing it against a database.
+func WithDryRunSpans() Option {
```

Comment:
Dry run spans sounds like an OpenTelemetry thing that creates some stub spans instead of normal spans, i.e. does not create spans at all. I will rename this to `DryRunTx` for a bit more clarity.